### PR TITLE
Add unit test cases for download_api_test.go&algorithm_test.go&atomiccount_test.go&httpuils_test.go 

### PR DESCRIPTION
**Pull request by Hellcatlk**
Signed-off-by: ZouYu <zouy.fnst@cn.fujitsu.com>

### Ⅰ. Describe what this PR did
 Add unit test case

### Ⅱ. Does this pull request fix one issue?

NONE

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)
NONE

### Ⅳ. Describe how to verify it
NONE

### Ⅴ. Special notes for reviews
NONE

### CODE REVIEW

File: dfget/util/algorithm_test.go [hunk 1]
```diff
@@ -52,4 +52,10 @@ func (s *AlgorithmTestSuite) TestShuffle(c *check.C) {
 		Shuffle(n, func(i, j int) { isRun++ })
 		c.Assert(isRun, check.Equals, n-1)
 	}
+
+	Shuffle(1<<31, func(i, j int) {
+		if 31-1 <= i {
```
Comment (lowzj) [inline dfget/util/algorithm_test.go:57]: Could you please add some description here? I'm not very clear about its meaning of this.

Comment (Hellcatlk) [reply dfget/util/algorithm_test.go:57]: Because if i let the loop complete, it will time out.
So i need to make the loop stop early.

Comment (Hellcatlk) [reply dfget/util/algorithm_test.go]: sorry, this is my mistake. I should be 1<<31-1.

File: pkg/atomiccount/atomiccount_test.go [hunk 2]
```diff
@@ -39,6 +39,9 @@ func (suite *AtomicCountUtilSuite) TestAdd(c *check.C) {
 
 	result := acount.Get()
 	c.Check(result, check.Equals, (int32)(12))
+
+	var nilAcount *AtomicInt
+	c.Check(nilAcount.Add(5), check.Equals, (int32)(0))
```
Comment (lowzj) [inline pkg/atomiccount/atomiccount_test.go:44]: It makes me confused why there is no panic when invoking the function of nil struct pointer. After a while, I understand: `nilAcount.Add(5)` equals `(*AtomicInt).Add(nilAcount, 5)` in GoLang, and the implementation of function `AtomicInt#Add` checks `nil` pointer before dereference it.

But is it a reasonable implementation?
* It's rare to check itself in a struct's own function. Why not let it panic directly?
* The behavior of the `AtomicInt` functions are inconsistent: `Set` doesn't check.
* Any value plus zero value should be equal to itself.

@Starnop WDYT

**Comment by lowzj** [comment]
Could you rebase your code and combine your commits into one commit?

**Comment by Hellcatlk** [comment]
@lowzj , Of course. Done.

**Pull request merged**